docs/cart-and-checkout-blocks/additional-checkout-fields.md

```diff
@@ -569,6 +566,426 @@ add_action(
+				},
+				"address_1": {
+					"type": "string",
+					"description": "Primary address line"
+				},
+				"address_2": {
+					"type": "string",
+					"description": "Secondary address line"
+				},
+				"city": {
+					"type": "string",
+					"description": "City name"
+				},
+				"state": {
+					"type": "string",
+					"description": "State or province, this will be the state code if it's a predefined list, for example: CA, TX, NY, etc, or the field value if it's a freeform state, for example: London."
+				},
+				"postcode": {
+					"type": "string",
+					"description": "Postal or ZIP code"
+				},
+				"country": {
+					"type": "string",
+					"description": "Country code (e.g., US, UK)"
+				},
+				"email": {
+					"type": "string",
+					"format": "email",
+					"description": "Email address"
+				},
+				"phone": {
+					"type": "string",
+					"description": "Phone number"
+				}
+			},
+			"additionalProperties": {
+				"type": "string",
+				"description": "Custom fields with namespace identifiers"
+			},
+			"patternProperties": {
+				"^[a-zA-Z0-9_-]+/[a-zA-Z0-9_-]+$": {
+					"type": "string",
+					"description": "Custom fields with namespace identifiers"
+				}
+			}
+		}
+	}
+}
+```
+
+</details>
+<!-- markdownlint-enable MD033 -->
+
+### Examples
+
+#### Required and visible field
+
+In this example we make the field required and visible only if local pickup is being used.
+
+```php
+'required' => [
+    "type" => "object",
+	"properties" => [
+		"cart" => [
+			"properties" => [
+				"prefers_collection" => [
+					"const" => true
+				]
+			]
+		]
+	]
+],
+'hidden' => [
+	"type" => "object",
+	"properties" => [
+		"cart" => [
+			"properties" => [
+				"prefers_collection" => [
+					"const" => false
+				]
+			]
+		]
+	]
+]
+```
+
+Notice that for hidden, we inverse the field, meaning, this field should only be hidden if `prefers_collection` is false, which is almost all cases except when it's selected. In the examples above, we used [the keyword `const`](https://ajv.js.org/json-schema.html#const).
+
+
+#### Validation schema example
+
+Validation is slightly different from conditional visibility and requirement. In validation, you will pass in a subset of schema (only applicable to your field), and its role is to validate the field and show any errors if there.
+
+In this example, we ensure that VAT is made up of a country code and 8-12 numbers.
+
+```php
+'validation' => [
+	"type" => "string",
+	"pattern" => "^[A-Z]{2}[0-9]{8,12}$"
+	"errorMessage" => "Please enter a valid VAT code with 2 letters for country code and 8-12 numbers."
+]
+```
+
+Validation can also be against other fields, for example, an alternative email field that shouldn't the current email:
+
+```php
+'validation' => [
+	"type" => "string",
+	"format" => "email",
+	"not" => [
+		"const" => ["$data", "0/customer/billing_address/email"]
+	]
+	"errorMessage" => "Please enter a valid VAT code with 2 letters for country code and 8-12 numbers."
```

Comment:
It would be good if this was updated to show an error message more relevant to the rule. I know it's just an example but it would help.